docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3942,18 +4037,26 @@ In order to support the above SDN use case, P4Runtime requires translation of
 port metadata values between the controller's space and the PSA device's space
 as needed. Such translation is enabled by identifying a P4 entity (match field,
 action parameter or header field) as being a PSA port metadata type. For this
-purpose, PSA defines the port metadata field type using special P4 types, namely
-`PortId_t` and `PortInHeader_t`, instead of standard P4 bitstrings. The P4Info
-for all P4 entities of the special PSA port types use a controller-defined
-32-bit type instead of the dataplane bitwidth defined in the P4 program. The
-following PSA port metadata types are defined in *psa.p4* for the PSA device in
-Switch 1.
+purpose, PSA defines the port metadata field type using special [user-defined P4
+types](#sec-user-defined-types), namely `PortId_t` and `PortInHeader_t`, instead
+of standard P4 bitstrings. The P4Info for all P4 entities of the special PSA
+port types use a controller-defined 32-bit type instead of the dataplane
+bitwidth defined in the P4 program. The following PSA port metadata types are
+defined in *psa.p4* for the PSA device in Switch 1.
```

Comment:
If I am reading this correctly, this is saying that all of the PSA custom width types will be 32 bits wide in P4Runtime messages.  I guess I had been assuming that the "*InHeader_t" bit widths in the latest working PSA spec would be the P4Runtime widths, and those are not all 32 bits wide.  The timestamp one doesn't even fit in 32 bits, but the rest of them do, so no big deal (except for the timestamp) if they all end up being 32 bits in P4Runtime messages.

Here is a link to the PSA spec Madoko source for quick reference: https://github.com/p4lang/p4-spec/blob/master/p4-16/psa/psa.p4#L75-L92

Note especially the comment block just before those types, which may be wrong and need updating, which I'd like to do when we decide the desired approach:
```
/* Note: All of the widths for types with `InHeader` in their name are
 * intended only to carry values of the corresponding types in packet
 * headers between a controller and a PSA device.  The widths are
 * intended to be large enough for all PSA devices, so that a
 * controller can fill in headers with these fields in a common way
 * for all PSA devices.
 *
 * All widths must be a multiple of 8, so that any subset of these
 * fields may be used in a single P4 header definition, even on P4
 * implementations that restrict headers to contain fields with a
 * total length that is a multiple of 8 bits. */
typedef bit<32> PortIdInHeader_t;
typedef bit<32> MulticastGroupInHeader_t;
typedef bit<16> CloneSessionIdInHeader_t;
typedef bit<8>  ClassOfServiceInHeader_t;
typedef bit<16> PacketLengthInHeader_t;
typedef bit<16> EgressInstanceInHeader_t;
typedef bit<64> TimestampInHeader_t;
```

Reply:
It may not be super clear the way Samar wrote it, but it does say "port types"; it only applies to `PortId_t` and `PortIdInHeader_t`.

Reply:
I think I was maybe reading a bit too quickly and missed the "port types", but they are mentioned and no need to change anything further here that I can see.